City-of-Helsinki/hel-fi-drupal-grants
# Feature/AU-1258: Application search with Search API

# [AU-1258](https://helsinkisolutionoffice.atlassian.net/browse/AU-1258)

## What was done

This pull request implements a new application search that has been implemented with the help of the Search API module.


## How to install

Make sure your instance is up and running on the correct branch.
  * `git checkout feature/AU-1258-search-api`
  * `make fresh`
  * `make drush-cr`

## How to test
1. Start by going to `/admin/config/search/path?search=&sort=asc&order=System path` and delete the old aliases for `/etsi-avustusta`. Now implement two new aliases for `/application-search` like shown in the picture below:
<img width="1673" alt="image" src="https://github.com/City-of-Helsinki/hel-fi-drupal-grants/assets/117807015/35d3bb4a-26fa-4730-83b2-e64519176994">

2. Now go to `/admin/config/search/search-api/index/search_index` and index all the service pages to the search index.

3. Next, go to `/fi/etsi-avustusta` and test that the search works and produces good results. Also review the search in both English and Swedish and check that the translations work.

3. Finally review the settings for the search index, mainly at `/admin/config/search/search-api/index/search_index/fields` and `/admin/config/search/search-api/index/search_index/processors`. Do the field and processor settings look alright? Do we need to adjust any of the boost options for the indexed fields?


[AU-1258]: https://helsinkisolutionoffice.atlassian.net/browse/AU-1258?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
